Bottom: 079d7bbe6129342c62546a42ccf45af677b9160f
Top:    8e858610108557f53e93dd7af44d25857fb890d0
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-07-18 09:09:30 -0700

Refresh of fork

---

diff --git a/src/preload.c b/src/preload.c
index 6cc0d72..e901d13 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -468,7 +468,7 @@ static int connect_fork(int socket, const struct sockaddr *addr, socklen_t addrl
 	if (ret)
 		return ret;
 
-	ret = real.read(fd, &msg, sizeof msg);
+	ret = real.recv(fd, &msg, sizeof msg, MSG_PEEK);
 	if ((ret != sizeof msg) || msg) {
 		fd_store(socket, fd, fd_normal);
 		return 0;
